### PR TITLE
First pass at Mortgage Performance Trends apps

### PR DESCRIPTION
**Pull request by contolini**
This is a first pass at migrating the Mortgage Performance JS into cfgov-refresh in preparation for upcoming user testing. The "apps" are a series of interdependent dropdowns, radio buttons and checkboxes that change the values of a line chart and a map of the U.S. At this stage it's just a copy-and-paste from the static site repo we've been using for rapid prototyping. Both the JS and HTML will be cleaned up in future PRs. The goal here is just to establish where the code will live.

The JS will only be included on pages that have [Mortgage Performance blocks](https://github.com/cfpb/cfgov-refresh/pull/3260) (which will be a couple pages in the `data-research` section of cf.gov).

## Additions

- Mortgage Performance JS into a new `MortgagePerformanceTrends/` organisms directory.

## Removals

-

## Changes

- @higs4281 I changed the `time_frame` context from `30` to `30-89` to match the API. 
- @sebworks I updated `uglifyjs-webpack-plugin` to have it play nicer with ES6. See the inline code comments.

## Testing

1. Set up the [mortgage performance API](https://github.com/cfpb/cfgov-refresh/pull/3189).
1. Create a [mortgage chart block and page](https://github.com/cfpb/cfgov-refresh/pull/3260).

## Notes

- Much of the heavy lifting (chart rendering and re-rendering) is done by cfpb-chart-builder. I published the development branch of that library under a new name (cfpb-chart-builder-canary) so that its new MP features can be pulled into cfgov-refresh's MP pages without having to constantly bump and release cfpb-chart-builder. For release we'll remove cfpb-chart-builder-canary and increment cfpb-chart-builder. Is this an okay approach?

## Todos

- Port unit tests over to cfgov-refresh.
- Add browser tests.
- Remove redundant JS (the line chart and map share a lot of the same code).
- All sorts of general clean up and bug fixing.

## Screenshots

screen shot 2017-08-29 at 10 52 43 am

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

File: cfgov/unprocessed/js/organisms/MortgagePerformanceTrends/apps/map.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    this.chart.highchart.chart.get( state.geo.id ).select();
+    this.chart.highchart.chart.get( state.geo.id ).zoomTo();
+    this.chart.highchart.chart.mapZoom( 5 );
+  }
+  if ( prevState.date !== state.date || prevState.geo.type !== state.geo.type ) {
+    store.dispatch( actions.startLoading() );
+    this.chart.update( {
+      source: `map-data/${ this.timespan }/${ _plurals[state.geo.type] }/${ state.date }`,
+      metadata: _plurals[state.geo.type]
+    } ).then( () => {
```
Comment (sebworks) [inline cfgov/unprocessed/js/organisms/MortgagePerformanceTrends/apps/map.js]: how does Babel handle promises?

**Comment by sebworks** [comment]
Looks good, and I like the reduxy stuff. Not sure on the usage of the underscore for private variables or the use of `$` for elements. I guess as long as we are consistent with it.

File: cfgov/unprocessed/js/organisms/MortgagePerformanceTrends/apps/map.js (hunk 1 above)
Comment (contolini) [reply cfgov/unprocessed/js/organisms/MortgagePerformanceTrends/apps/map.js]: Mmmm good question. I know cfpb-chart-builder uses promises and it supports IE9+. I'll do some investigating before merging this PR.

File: cfgov/unprocessed/js/organisms/MortgagePerformanceTrends/utils.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+var ajax = require( 'xdr' );
```
Comment (contolini) [inline cfgov/unprocessed/js/organisms/MortgagePerformanceTrends/utils.js]: FYI @sebworks I'll replace this third party xhr library with [ours](https://github.com/cfpb/cfgov-refresh/blob/6b7f7d005a0ce189e84151a31801077dc76e7897/cfgov/unprocessed/js/modules/util/ajax-request.js) in a future PR.

File: cfgov/unprocessed/js/organisms/MortgagePerformanceTrends/stores/store.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+class Store {
```
Comment (sebworks) [inline cfgov/unprocessed/js/organisms/MortgagePerformanceTrends/stores/store.js]: We should probably move this to utilities at some point. I could see us reusing this in the future.

Comment (contolini) [reply cfgov/unprocessed/js/organisms/MortgagePerformanceTrends/stores/store.js]: Cool I'll sprinkle some `TODO`s throughout the code for future PRs.

**Pull request merged**